### PR TITLE
Add async sgd document

### LINKED ISSUE #8268: Async SGD document need to be updated.
As mentioned in https://github.com/PaddlePaddle/Paddle/issues/7108#issuecomment-354717422. Current documentation about V1/V2 Async-SGD need to be updated. @shanyi15

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #8268

### CODE REVIEW

File: python/paddle/trainer_config_helpers/optimizers.py [hunk 1]
```diff
@@ -365,7 +365,8 @@ def settings(batch_size,
              regularization=None,
              is_async=False,
              model_average=None,
-             gradient_clipping_threshold=None):
+             gradient_clipping_threshold=None,
```
Comment (typhoonzero) [inline python/paddle/trainer_config_helpers/optimizers.py]: Seems these changes broke the CI unit tests

Comment (Yancey1989) [reply python/paddle/trainer_config_helpers/optimizers.py]: Done.

File: doc/howto/cluster/cmd_argument_cn.md [hunk 2]
```diff
@@ -133,3 +140,20 @@ test.txt-00002
 
 - `train_data_dir`：包含训练数据的目录，可以是从分布式存储挂载过来的，也可以是在任务启动前下载到本地的。
 - `test_data_dir`：包含测试数据集的目录。
+
+## 异步 SGD 更新
+
+您可以修改 `optimize` 的参数来支持异步SGD更新，示例如下：
```
Comment (wangkuiyi) [inline doc/howto/cluster/cmd_argument_cn.md]: 下文里没有一个叫做 optimize 的参数呀？

File: doc/howto/cluster/cmd_argument_en.md [hunk 3]
```diff
@@ -138,3 +141,20 @@ Your workspace may looks like:
 
 - `train_data_dir`: containing training data. Mount from storage service or copy trainning data to here.
 - `test_data_dir`: containing testing data.
+
+## Async SGD Update
+
+You can modify the parameters of `optimize` to support Async SGD update, for example:
```
Comment (wangkuiyi) [inline doc/howto/cluster/cmd_argument_en.md]: I don't see a parameter named `optimize` in the following code snippet. Do you mean

> We can set some parameters of the optimizer to make it support async SGD update.  For example, we can set the `is_async` and `async_lagged_grad_discard_ratio` of the `AdaGrad` optimizer:

File: python/paddle/trainer_config_helpers/optimizers.py [hunk 4]
```diff
@@ -396,6 +397,10 @@ def settings(batch_size,
                                         value larger than some value, will be
                                         clipped.
     :type gradient_clipping_threshold: float
+    :param async_lagged_grad_discard_ratio: async sgd gradient commit control,
```
Comment (wangkuiyi) [inline python/paddle/trainer_config_helpers/optimizers.py]: sgd => SGD

File: python/paddle/trainer_config_helpers/optimizers.py [hunk 5]
```diff
@@ -396,6 +397,10 @@ def settings(batch_size,
...
+          when async_lagged_grad_discard_ratio * num_gradient_servers commit passed, 
+          current async gradient will be discard silently.
```
Comment (wangkuiyi) [inline python/paddle/trainer_config_helpers/optimizers.py]: > will be discard silently

==>

> is discarded

Comment (wangkuiyi) [inline python/paddle/trainer_config_helpers/optimizers.py]: > current sgd gradient

==>

> the current SGD gradient

File: doc/howto/cluster/cmd_argument_en.md (hunk 3 above)
Comment (Yancey1989) [reply doc/howto/cluster/cmd_argument_en.md]: Thanks, this description is clearer.

File: python/paddle/trainer_config_helpers/optimizers.py (hunk 4 above)
Comment (Yancey1989) [reply python/paddle/trainer_config_helpers/optimizers.py]: Done.

File: python/paddle/trainer_config_helpers/optimizers.py (hunk 5 above)
Comment (Yancey1989) [reply python/paddle/trainer_config_helpers/optimizers.py]: Done.

Comment (Yancey1989) [reply python/paddle/trainer_config_helpers/optimizers.py]: Done.

File: doc/howto/cluster/cmd_argument_cn.md (hunk 2 above)
Comment (Yancey1989) [reply doc/howto/cluster/cmd_argument_cn.md]: Done as comment https://github.com/PaddlePaddle/Paddle/pull/8474#discussion_r170041097.

**Pull request merged**